Lib/pyclbr.py

```diff
@@ -160,17 +160,20 @@ def _readmodule(module, path, inpackage=None):
     else:
         search_path = path + sys.path
     spec = importlib.util._find_spec_from_path(fullmodule, search_path)
+    if spec is None:
+        raise ModuleNotFoundError(f"no module named {fullmodule!r}", name=fullmodule)
     _modules[fullmodule] = tree
     # Is module a package?
-    if spec.submodule_search_locations is not None:
+    if getattr(spec, 'submodule_search_locations', None) is not None:
```

Comment:
Currently I don't see the need of using `getattr` here. `spec` would be an instance of `importlib.machinery.ModuleSpec` and the `submodule_search_locations` attribute is always present and can either be a list or None (`L: 348 @ Lib/importlib/_boostrap.py`).

Was this done with future proofing in mind? because a change to `ModuleSpec.submodule_search_locations` availability would ideally come with depreciation warnings giving plenty of time for change.

Reply:
I'm just being paranoid here. It's more if someone screwed up and put something other than an actual `ModuleSpec` in the object's `__spec__` attribute (you might be surprised the random stuff people cram in places and the bug reports we get about it).

Basically since it isn't catastrophic if that attribute doesn't exist I decided to hedge my bets.